packages/bot-web-ui/src/utils/xml-dom-quick-strategy.ts

```diff
@@ -20,7 +20,7 @@ export const addDynamicBlockToDOM = (
         shadow_block.appendChild(field_block);
         block.appendChild(shadow_block);
 
-        const amount_block = strategy_dom.querySelector('value[name="AMOUNT"]');
+        const amount_block = strategy_dom?.querySelector('value[name="AMOUNT"]');
```

Comment:
@mayuran-deriv is it necessary to use optional chaining here? `strategy_dom` is a required argument. If this one is not passed, the next lines (from lines 24 to 29) will not be covered.

Reply:
hey man in my test case im checking for the null value so thats why i added this.


  it('should not modify DOM if strategy_dom is null', () => {
        const initialDOM = document.body.innerHTML;
        addDynamicBlockToDOM('testBlock', 'testValue', 'digits', null as unknown as HTMLElement);
        expect(document.body.innerHTML).toBe(initialDOM);
    });